### PR TITLE
feat #14-15: Create database and class for connecting to it

### PR DESCRIPTION
**Pull request by Kayi1500**
Add database.sql for creating database and table containing commits.

Implement MysqlDatabase class for handling connection to database and retrieving/inserting commits.

### CODE REVIEW

**Review by grundb — COMMENTED** [review]
Looks good but we should probably find a way to pass Travis CI, also the test passed boolean was removed in the main branch

File: src/main/java/MysqlDatabase.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            System.out.println("Connection is null!");
+        }
+        return false;
+    }
+
+    /**
+     * Selects all rows from database
+     * @return Returns a list with all of the rows read as CommitStructure objects
+     */
+    public static List<CommitStructure> selectAllCommits() throws SQLException {
```
Comment (grundb) [inline src/main/java/MysqlDatabase.java:68]: If connection is used as parameter, we can test these better

Comment (Kayi1500) [reply src/main/java/MysqlDatabase.java]: Fixed in commit https://github.com/DD2480-2021-group-22/dd2480-lab-2/pull/24/commits/0aa78b81fc4b991068de267a30ba058e373b13e4

File: src/main/java/CommitStructure.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/**
+ * Class with all the columns for table "commit" and their getters/setters. Simplifies fetching results from database
+ */
+public class CommitStructure {
+    private String commitID , buildDate , buildLogs;
+    private boolean testResult, buildResult;
```
Comment (grundb) [inline src/main/java/CommitStructure.java:6]: testResult not used

File: src/test/java/DatabaseTest.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+     */
+    @Test
+    public void testSelectingAllRowsWhenNotEmpty() throws SQLException {
+        // Arrange
+        CommitStructure commit = getSampleCommit();
+        // Act
+        mysqlDatabase.insertCommitToDatabase(commit);
+        List<CommitStructure> commits = mysqlDatabase.selectAllCommits();
+        // Assert
+        assertFalse(commits.isEmpty());
```
Comment (kittytinythai) [inline src/test/java/DatabaseTest.java:92]: Add asserts to check that the inserted row is equal the selected one.

**Review by kittytinythai — CHANGES_REQUESTED** [review]

File: src/main/java/MysqlDatabase.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        this.connection = connection;
+    }
+
+    /**
+     * Creates a Connection object and connects it to localhost mysql server
+     * @return Connection object
+     */
+    private Connection connectToDB() {
+        try{
+            //Try getting an object from class com.mysql.jdbc.Driver
```
Comment (grundb) [inline src/main/java/MysqlDatabase.java:31]: add source for sample code

**Review by grundb — APPROVED** [review]

**Pull request merged**